src/test/java/TypewiseAlert/TypewiseAlertTest.java

```diff
@@ -1,15 +1,48 @@
 package TypewiseAlert;
 
+
 import static org.junit.Assert.assertTrue;
 
 import org.junit.Test;
 
-public class TypewiseAlertTest 
-{
-    @Test
-    public void infersBreachAsPerLimits()
-    {
-      assertTrue(TypewiseAlert.inferBreach(12, 20, 30) ==
-        TypewiseAlert.BreachType.TOO_LOW);
-    }
+
+import TypewiseAlert.BatteryCharacter;
+import TypewiseAlert.TypewiseAlert;
+import TypewiseAlert.constants.AlertTarget;
+import TypewiseAlert.constants.BreachType;
+import TypewiseAlert.constants.CoolingType;
+
+public class TypewiseAlertTest {
+	
+	@Test
+	public void infersBreachAsPerLimits() {
+		assertTrue(BreachType.inferBreach(12, 20, 30) == BreachType.TOO_LOW);
+		assertTrue(BreachType.inferBreach(40, 30, 40) == BreachType.NORMAL);
+		assertTrue(BreachType.inferBreach(55, 45, 50) == BreachType.TOO_HIGH);
+	}
+	
+	@Test
+	public void verifyBreachType() {
+		assertTrue(TypewiseAlert.classifyTemperatureBreach(CoolingType.PASSIVE_COOLING, -10) == BreachType.TOO_LOW);
+		assertTrue(TypewiseAlert.classifyTemperatureBreach(CoolingType.MED_ACTIVE_COOLING, 15) == BreachType.NORMAL);
+		assertTrue(TypewiseAlert.classifyTemperatureBreach(CoolingType.HI_ACTIVE_COOLING, 55) == BreachType.TOO_HIGH);
+	}
+	 @Test
+	 public void verifyEmailHighAlertSent() {
+		 BatteryCharacter batteryChar = new BatteryCharacter(CoolingType.HI_ACTIVE_COOLING, "bosch");
+		 assertTrue(TypewiseAlert.checkAndAlert(AlertTarget.TO_EMAIL, batteryChar, 55) == 1);
+		 
+	 }
+	 
+	 @Test
+	 public void verifyEmailLowAlertSent() {
+		 BatteryCharacter batteryChar = new BatteryCharacter(CoolingType.PASSIVE_COOLING, "bosch");
+		 assertTrue(TypewiseAlert.checkAndAlert(AlertTarget.TO_EMAIL, batteryChar, -10) == 1);
+	 }
+	 
```

Comment:
What if you had to test the content of the email or the message to the controller, without actually having to see it in an inbox or hardware? How far can you go with unit tests?